### DISCUSSION TITLE
Migration to new version

### DISCUSSION THREAD

**Discussion by gitalexch** [kheeklab/privacyidea-docker#146] [Ideas]
Hi, I have a question about running a new version of PI first time. There are migrations that should be run if a new non-container version of PI is started at first time, but in case of protection current database(if a new version was run by mistake) it seems to be useful to make an env variable to disable (with default "disable" value) or enable automatic migration, i.e.  PI_UPGRADE=false.

So my question is about possibility to upgrade PI automatically and protect from auto upgrade by mistake. Is automatic upgrade supported by current version and is possible to add this function and additional protection?

Seems that the same problem was discussed https://github.com/Khalibre/privacyidea-docker/issues/65 but there was no result...

**Comment by mr-ssd** [comment]
@gitalexch thank for raising the idea. The discus #65 was my idea to auto upgrade the database using `pi-manage` cli but at that time there a issue in `pi-manage` https://github.com/privacyidea/privacyidea/issues/3570 which fixed in version 3.9. 

> So my question is about possibility to upgrade PI automatically and protect from auto upgrade by mistake. Is automatic upgrade supported by current version and is possible to add this function and additional protection?

yes it possible with your propose environment variable control. Unfortunately the current version is not supported but it is possible to add and might not complex.

You can submit a PR for that if you have an idea and we can work together about this.

Cheer!

**Reply by gitalexch** [reply]
Have added PR, but it's only DRAFT not ready for merging. The check of availability  persistent volume for backup should be added(I don't know how to do this better) and/or maybe some more additional  protections/warnings and logs.
https://github.com/Khalibre/privacyidea-docker/pull/158

**Reply by mr-ssd** [reply]
thx @gitalexch
